queries/sql/highlights.scm

```diff
@@ -27,13 +27,64 @@
 (term
   alias: (identifier) @variable)
 
+((term
+   value: (cast
+    name: (identifier) @function.call
+    parameter: [(literal)]?)))
+
 (literal) @string
 (comment) @comment
 (marginalia) @comment
 
 ((literal) @number
  (#lua-match? @number "^%d+$"))
 
+((literal) @float
+ (#lua-match? @float "^[-]?%d*.?%d*$"))
```

Comment:
```suggestion
 (#lua-match? @float "^[-+]?%d*\.%d*$"))
```

The `.` must be present so we need to remove the `?` and I think make it `\.` so it matches a literal `.` character and not any character.

Something like `+1.1` might be valid too so adding the `+` to the start of the regex would be good.

Reply:
The `+` is not supported by the grammar at the moment. I will leave this out and make the other changes.